elastic/kibana
# [Index management] Link component templates

Closes https://github.com/elastic/kibana/issues/183988 and https://github.com/elastic/kibana/issues/183987

## Summary
This PR makes the component templates list shown in index template details flytout into a list of links. If any of these component templates is a `@custom` template that doesnt yet exist a callout will be shown asking the user if he wants to create it. I thought about doing it this way to avoid having to fetch data about the custom template first and decide then if we redirect to details flyout or to creation.

##### How to test

* Start up kibana and es
* Navigate to Stack management -> Index Management -> Index templates
* Click `logs-apm.error@template` to see the flyout details
* Verify that:
  * Component templates are now links that navigate to the respecitve component template flyout
  * Clicking on a `@custom` component template that is missing shows an empty flyout with a prompt saying that it doesnt yet exist and if you want to create it. Upon clicking that link it should redirect the user to the component creation flow with a pre-populated component name


https://github.com/elastic/kibana/assets/1191206/7c70e640-5144-4951-84a2-23ecf1211afb
